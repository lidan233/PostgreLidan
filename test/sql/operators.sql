-- comparison
SELECT '120'::lidan > '3c'::lidan;
SELECT '120'::lidan >= '3c'::lidan;
SELECT '120'::lidan < '3c'::lidan;
SELECT '120'::lidan <= '3c'::lidan;
SELECT '120'::lidan <> '3c'::lidan;
SELECT '120'::lidan = '3c'::lidan;

-- comparison equals
SELECT '120'::lidan > '120'::lidan;
SELECT '120'::lidan >= '120'::lidan;
SELECT '120'::lidan < '120'::lidan;
SELECT '120'::lidan <= '120'::lidan;
SELECT '120'::lidan <> '120'::lidan;
SELECT '120'::lidan = '120'::lidan;

-- comparison negation
SELECT NOT '120'::lidan > '120'::lidan;
SELECT NOT '120'::lidan >= '120'::lidan;
SELECT NOT '120'::lidan < '120'::lidan;
SELECT NOT '120'::lidan <= '120'::lidan;
SELECT NOT '120'::lidan <> '120'::lidan;
SELECT NOT '120'::lidan = '120'::lidan;

--commutator and negator
BEGIN;
CREATE TABLE lidan_test AS
SELECT i::lidan as val FROM generate_series(1,10000) i;
CREATE INDEX ON lidan_test(val);
ANALYZE;
SET enable_seqscan TO off;
EXPLAIN (COSTS OFF) SELECT * FROM lidan_test where NOT val < 'c1';
EXPLAIN (COSTS OFF) SELECT * FROM lidan_test where NOT 'c1' > val;
EXPLAIN (COSTS OFF) SELECT * FROM lidan_test where 'c1' > val;
-- hash aggregate
SET enable_seqscan TO on;
EXPLAIN (COSTS OFF) SELECT val, COUNT(*) FROM lidan_test GROUP BY 1;
ROLLBACK;