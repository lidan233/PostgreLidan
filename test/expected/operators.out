-- comparison
SELECT '120'::lidan > '3c'::lidan;
 ?column? 
----------
 t
(1 row)

SELECT '120'::lidan >= '3c'::lidan;
 ?column? 
----------
 t
(1 row)

SELECT '120'::lidan < '3c'::lidan;
 ?column? 
----------
 f
(1 row)

SELECT '120'::lidan <= '3c'::lidan;
 ?column? 
----------
 f
(1 row)

SELECT '120'::lidan <> '3c'::lidan;
 ?column? 
----------
 t
(1 row)

SELECT '120'::lidan = '3c'::lidan;
 ?column? 
----------
 f
(1 row)

-- comparison equals
SELECT '120'::lidan > '120'::lidan;
 ?column? 
----------
 f
(1 row)

SELECT '120'::lidan >= '120'::lidan;
 ?column? 
----------
 t
(1 row)

SELECT '120'::lidan < '120'::lidan;
 ?column? 
----------
 f
(1 row)

SELECT '120'::lidan <= '120'::lidan;
 ?column? 
----------
 t
(1 row)

SELECT '120'::lidan <> '120'::lidan;
 ?column? 
----------
 f
(1 row)

SELECT '120'::lidan = '120'::lidan;
 ?column? 
----------
 t
(1 row)

-- comparison negation
SELECT NOT '120'::lidan > '120'::lidan;
 ?column? 
----------
 t
(1 row)

SELECT NOT '120'::lidan >= '120'::lidan;
 ?column? 
----------
 f
(1 row)

SELECT NOT '120'::lidan < '120'::lidan;
 ?column? 
----------
 t
(1 row)

SELECT NOT '120'::lidan <= '120'::lidan;
 ?column? 
----------
 f
(1 row)

SELECT NOT '120'::lidan <> '120'::lidan;
 ?column? 
----------
 t
(1 row)

SELECT NOT '120'::lidan = '120'::lidan;
 ?column? 
----------
 f
(1 row)

--commutator and negator
BEGIN;
CREATE TABLE lidan_test AS
SELECT i::lidan as val FROM generate_series(1,10000) i;
CREATE INDEX ON lidan_test(val);
ANALYZE;
SET enable_seqscan TO off;
EXPLAIN (COSTS OFF) SELECT * FROM lidan_test where NOT val < 'c1';
                        QUERY PLAN                        
----------------------------------------------------------
 Index Only Scan using lidan_test_val_idx on lidan_test
   Index Cond: (val >= 'c1'::lidan)
(2 rows)

EXPLAIN (COSTS OFF) SELECT * FROM lidan_test where NOT 'c1' > val;
                        QUERY PLAN                        
----------------------------------------------------------
 Index Only Scan using lidan_test_val_idx on lidan_test
   Index Cond: (val >= 'c1'::lidan)
(2 rows)

EXPLAIN (COSTS OFF) SELECT * FROM lidan_test where 'c1' > val;
                        QUERY PLAN                        
----------------------------------------------------------
 Index Only Scan using lidan_test_val_idx on lidan_test
   Index Cond: (val < 'c1'::lidan)
(2 rows)

-- hash aggregate
SET enable_seqscan TO on;
EXPLAIN (COSTS OFF) SELECT val, COUNT(*) FROM lidan_test GROUP BY 1;
          QUERY PLAN           
-------------------------------
 HashAggregate
   Group Key: val
   ->  Seq Scan on lidan_test
(3 rows)

ROLLBACK;
